Source/Tests/LinuxSyscalls/x64/Memory.cpp

```diff
@@ -25,17 +25,21 @@ namespace FEX::HLE::x64 {
       uint64_t Result = ::munmap(addr, length);
       if (Result != -1) {
         FEXCore::Context::RemoveNamedRegion(Thread->CTX, (uintptr_t)addr, length);
+        FEXCore::Context::FlushCodeRange(Thread, (uintptr_t)addr, (uintptr_t)addr + length);
       }
       SYSCALL_ERRNO();
     });
 
     REGISTER_SYSCALL_IMPL_X64(mmap, [](FEXCore::Core::InternalThreadState *Thread, void *addr, size_t length, int prot, int flags, int fd, off_t offset) -> uint64_t {
       static FEXCore::Config::Value<bool> AOTIRLoad(FEXCore::Config::CONFIG_AOTIR_LOAD, false);
       uint64_t Result = reinterpret_cast<uint64_t>(::mmap(addr, length, prot, flags, fd, offset));
-      if (Result != -1 && !(flags & MAP_ANONYMOUS)) {
-        auto filename = get_fdpath(fd);
+      if (Result != -1) {
+        if (!(flags & MAP_ANONYMOUS)) {
+          auto filename = get_fdpath(fd);
 
-        FEXCore::Context::AddNamedRegion(Thread->CTX, Result, length, offset, filename);
+          FEXCore::Context::AddNamedRegion(Thread->CTX, Result, length, offset, filename);
+        }
+        FEXCore::Context::FlushCodeRange(Thread, (uintptr_t)Result, (uintptr_t)Result + length);
```

Comment:
Do we really need to flush the code range?

Nothing was previously here, and if something was here earlier, we *should* have flushed it on unmap

Reply:
Flushing is required if `MAP_FIXED` is used.

Reply:
@phire flushing is fast when there's no backing code, so i've added it just to be sure. We don't catch all mmaps yet (namely, we don't catch mmaps from the elf iinterpreter), though I don't think anything will unload that